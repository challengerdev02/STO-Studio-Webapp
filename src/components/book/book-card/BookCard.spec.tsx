import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BookCard } from './index';

describe('BookCard component', function () {
  afterEach(cleanup);
  let props = {
    title: 'This is a test title',
    author: 'John Doe',
    description: 'This is metacomic',
    imageUrl: '/book-cover.png',
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<BookCard {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should test the BookCard component has a display image ', () => {
    const { getByTestId } = setup();
    const displayImg = getByTestId('display-image');
    expect(displayImg).toBeInTheDocument();
    expect(displayImg).toHaveAttribute('src', props.imageUrl);
  });

  it('should test the BookCard component has a title displayed ', () => {
    const { getByTestId } = setup();
    expect(getByTestId('book-title')).toBeInTheDocument();
  });

  it('should test the BookCard component has an author text displayed ', () => {
    const { getByTestId } = setup();
    expect(getByTestId('book-author')).toBeInTheDocument();
  });

  it('should test the BookCard component has a description displayed ', () => {
    const { getByTestId } = setup();
    expect(getByTestId('book-description')).toBeInTheDocument();
  });

  it('should test the Sell button is painted on screen ', () => {
    const { getByRole } = setup();
    const button = getByRole('button', { name: /sell/i });
    expect(button).toBeInTheDocument();
  });

  it('should test the BookCard component has an options component displayed ', () => {
    const { getByTestId } = setup();
    expect(getByTestId('options')).toBeInTheDocument();
  });
});
