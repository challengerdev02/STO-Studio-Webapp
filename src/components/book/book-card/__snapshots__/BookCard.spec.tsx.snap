// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard component should match with snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL fbVEoT"
    data-testid="create-book"
  >
    <div
      class="sc-gsDKAQ isItYx"
    >
      <div
        class="sc-dkPtRN ivtszv"
      >
        <img
          class="sc-hKwDye hMpCoO"
          data-testid="display-image"
          src="/book-cover.png"
        />
      </div>
      <div
        class="sc-eCImPb hmCnvY"
      >
        <div
          class="sc-furwcr ghJnqx"
        >
           
          <div
            class="sc-jRQBWg hgvvMF"
            data-testid="book-title"
          >
            This is a test title
          </div>
          <div
            class="sc-gKclnd Yeocv"
            data-testid="book-author"
          >
            by @
            John Doe
          </div>
          <div
            aria-label="This is metacomic"
            class="ant-typography ant-typography-ellipsis sc-iCfMLu cqVfIu"
            data-testid="book-description"
          >
            This is metacomic
            <span
              aria-hidden="true"
              style="position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; word-break: keep-all; white-space: nowrap;"
            >
              lg
            </span>
            <span
              aria-hidden="true"
              style="position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; width: 0px; white-space: normal; margin: 0px; padding: 0px;"
            >
              <span
                aria-hidden="true"
              >
                ...
              </span>
              <a
                aria-label="Expand"
                class="ant-typography-expand"
              >
                <span
                  class="read_more"
                >
                  Read More
                </span>
              </a>
            </span>
          </div>
        </div>
        <div
          class="sc-pVTFL dTSExf"
        >
          <button
            class="ant-btn ant-btn-default sc-jrQzAO fQSWGu"
            type="button"
          >
            <span>
              Sell
            </span>
          </button>
          <div
            class="sc-kDTinF jBCLks ant-dropdown-trigger sc-crHmcD"
            data-testid="options"
          >
            <span
              class="sc-iqseJM jFUQWc"
            >
              ...
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
